projects/packages/forms/src/contact-form/class-contact-form-plugin.php

```diff
@@ -1863,9 +1863,54 @@ public function download_feedback_as_csv() {
 		}
 
 		fclose( $output ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_fclose
+
+		$this->record_tracks_event( 'forms_export_responses', array( 'format' => 'csv' ) );
```

Comment:
I would recommend checking that the user has agreed to the ToS before launching the tracking. We did not necessarily need to do it in #29225 because the export to Google Sheets is only available on sites that are connected to WordPress.com (and thus where the admin has accepted the ToS), but here, the export to CSV is also available on offline sites, not connected to WordPress.com, who do not want and haven't agreed to be tracked or to have their activity recorded on WordPress.com.

In some other packages, we have a method like this one that can be helpful here:
https://github.com/Automattic/jetpack/blob/88e6782e6de6e5627af18781481a9f8a79d4302f/projects/packages/my-jetpack/src/class-initializer.php#L126-L136

Reply:
>  but here, the export to CSV is also available on offline sites, not connected to WordPress.com, who do not want and haven't agreed to be tracked or to have their activity recorded on WordPress.com.

@jeherve  Quick question: I was assuming you are only able to do the export via Jetpack Forms?
No matter if you are on WP.com or a self-hosted site, you need to have Jetpack installed, being able to access the JP form blocks, or the "Feedback>Form Responses" area in WP-Admin.


<img width="953" alt="image" src="https://user-images.githubusercontent.com/5120357/223131282-3cb610bb-440a-4b1e-a6c2-ca1ff81afec2.png">

Above, JN site, no Jetpack installed. No "Feedback" in WP-Admin menu on left, install of Jetpack needs ToS agreement.

Reply:
Try going to Settings > Jetpack Constants; you'll be able to enable Jetpack Offline mode there. Once it is enabled, you will be able to go to Jetpack > Settings to enable the Forms feature (and thus to export to csv).

Reply:
I just updated the PR added the connection check from [here](https://github.com/Automattic/jetpack/blob/6a7f974d7520963082f846fbae7bdf320a86e8b7/projects/packages/forms/src/contact-form/class-admin.php#L275) to make sure the user has connected to WordPress.com.

It wasn't enough to check the Terms_Of_Service(), as above, as that still returned true if I clicked through the "Set up Jetpack" button, but didn't connect.